proto/src/domain/nullifiers.rs

```diff
@@ -0,0 +1,41 @@
+use miden_objects::{crypto::merkle::SmtProof, Digest};
+
+use crate::{
+    errors::{MissingFieldHelper, ParseError},
+    generated::responses::NullifierBlockInputRecord,
+};
+
+// NULLIFIER INPUT RECORD
+// ================================================================================================
+
+#[derive(Clone, Debug)]
+pub struct NullifierWitness {
+    pub nullifier: Digest,
+    pub proof: SmtProof,
+}
```

Comment:
The type of `nullifier` should probably be `Nullifier` rather than `Digest` (and this is relevant to all other similar places).